Tests/eppo/AssignmentCacheTests.swift

```diff
@@ -15,33 +15,66 @@ final class AssignmentCacheTests: XCTestCase {
         cache = nil
         super.tearDown()
     }
-    
-    func testSetAndGet() {
-        let key = "testKey"
-        let value = "testValue"
-        XCTAssertNil(cache.get(key: key), "Cache should return nil for a key that has not been set.")
-        cache.set(key: key, value: value)
-        XCTAssertEqual(cache.get(key: key), value, "Cache should return the value that was set for a key.")
-    }
-    
-    func testHas() {
-        let key = "testKey"
-        XCTAssertFalse(cache.has(key: key), "Cache should return false for a key that has not been set.")
-        cache.set(key: key, value: "testValue")
-        XCTAssertTrue(cache.has(key: key), "Cache should return true for a key that has been set.")
+
+    func testOscillatingAllocation() {
+        let key1 = AssignmentCacheKey(
+            subjectKey: "Math", 
+            flagKey: "TestFlag",
+            allocationKey: "A1", // initial
+            variationKey: EppoValue(value: "VariationA", type: EppoValueType.String).toHashedString()
+        )
+        let key2 = AssignmentCacheKey(
+            subjectKey: "Math", 
+            flagKey: "TestFlag",
+            allocationKey: "A2", // changes
+            variationKey: EppoValue(value: "VariationA", type: EppoValueType.String).toHashedString()
+        )
+
+        cache.setLastLoggedAssignment(key: key1)
+        XCTAssertTrue(cache.hasLoggedAssignment(key: key1))
+        XCTAssertFalse(cache.hasLoggedAssignment(key: key2))
```

Comment:
This assertion is a stand-in for whether the event would be logged. Do you think it is sufficient to test this behavior?

Reply:
LGTM